# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webcodecs/DecoderTemplate.cpp
# Commit: e903ac081227
# Full Hash: e903ac0812273c476426eba3462c4bb2977c4f2f
# Author: Paul Adenot <paul@paul.cx>
# Date: 2023-12-23 09:52:03
# Regressor Bug: 1749047
# File Overlap Count: 3
# Description:
#   Bug 1749047 - Move and deduplicate some utility functions to WebCodecUtils. r=chunmin
#   
#   They will be reused for the encoding side.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D195699
# ==============================================================================

diff -r 029f5470314e -r e903ac081227 dom/media/webcodecs/DecoderTemplate.cpp
--- a/dom/media/webcodecs/DecoderTemplate.cpp	Fri Dec 22 21:39:04 2023 +0000
+++ b/dom/media/webcodecs/DecoderTemplate.cpp	Fri Dec 22 21:39:05 2023 +0000
@@ -137,8 +137,9 @@
   LOG("%s %p, Configure: codec %s", DecoderType::Name.get(), this,
       NS_ConvertUTF16toUTF8(aConfig.mCodec).get());
 
-  if (!DecoderType::Validate(aConfig)) {
-    aRv.ThrowTypeError("config is invalid");
+  nsCString errorMessage;
+  if (!DecoderType::Validate(aConfig, errorMessage)) {
+    aRv.ThrowTypeError(nsPrintfCString("config is invalid: %s", errorMessage.get()));
     return;
   }
 